### PR TITLE
Fix launching extension in safari

### LINKED ISSUE #10621: Jupyter can not get activated on Safari
```
Activating extension 'ms-toolsai.jupyter' failed: Invalid regular expression: invalid group specifier name.
```

since Safari doesn't support [lookbehind](https://caniuse.com/js-regexp-lookbehind).

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #10621

### CODE REVIEW

File: build/ci/postInstall.js [hunk 1]
```diff
@@ -125,7 +126,78 @@ function fixJupyterLabRenderers() {
...
+}
+function fixLuminoPolling() {
+    const warnings = [];
+    ['node_modules/@lumino/polling/dist/index.es6.js'].forEach((file) => {
+        const filePath = path.join(__dirname, '..', '..', file);
+        if (!fs.existsSync(filePath)) {
+            return;
+        }
+        const textToReplace = `: setImmediate;`;
+        const textToReplaceWith = `: typeof setImmediate === 'function' ? setImmediate : setTimeout;`;
```
Comment (DonJayamanne) [inline build/ci/postInstall.js:159]: Will submit PR upstream, `setImmediate` only runs in node. however we run our extensions in webworker, and jupyter code is written to run either in browser or node.

File: build/ci/postInstall.js [hunk 2]
```diff
@@ -125,7 +126,78 @@ function fixJupyterLabRenderers() {
...
+function fixStripComments() {
+    const file = 'node_modules/strip-comments/lib/languages.js';
+    const filePath = path.join(__dirname, '..', '..', file);
+    if (!fs.existsSync(filePath)) {
+        return;
+    }
+    const contents = dedent`
+    'use strict';
+
+    exports.python = {
```
Comment (DonJayamanne) [inline build/ci/postInstall.js:188]: Other reg exes are not required and that causes issues in loading extension.

File: build/launchWeb.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { launch } = require('./launchWebUtils');
```
Comment (DonJayamanne) [inline build/launchWeb.js:4]: Added ability to launch extension in vscode web locally

File: src/kernels/ipywidgets/baseIPyWidgetScriptManager.ts [hunk 4]
```diff
@@ -66,7 +66,7 @@ export function extractRequireConfigFromWidgetEntry(baseUrl: Uri, widgetFolderNa
     // the config entry is js, and not json.
     // We cannot eval as thats dangerous, and we cannot use JSON.parse either as it not JSON.
     // Lets just extract what we need.
-    configStr = stripComments(configStr);
+    configStr = stripComments(configStr, { language: 'python' });
```
Comment (DonJayamanne) [inline src/kernels/ipywidgets/baseIPyWidgetScriptManager.ts:69]: Be explicit about the fact that we're only stripping js code comments

File: build/ci/postInstall.js (hunk 1 above)
Comment (DonJayamanne) [reply build/ci/postInstall.js]: will look into (tomorrow) using the setimmediate pollyfill npm package as an alternative to modifying such scripts.

File: build/ci/postInstall.js [hunk 5]
```diff
@@ -125,7 +126,77 @@ function fixJupyterLabRenderers() {
     }
 }
 
+function fixJupyterLabFuture() {
```
Comment (IanMatthewHuff) [inline build/ci/postInstall.js:129]: Do we need like, a comment here? Or a link to the PR upstream?

**Review by IanMatthewHuff — APPROVED** [review]
Could maybe use a comment in postInstall.js

**Review by rchiodo — APPROVED** [review]

**Comment by DonJayamanne** [comment]
@IanMatthewHuff @rchiodo please re-review, updated the solution

**Review by rchiodo — APPROVED** [review]

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**